src/coroutine/cancel.rs

```diff
@@ -68,9 +68,9 @@ impl Future for Cancelled<'_> {
 pub struct ThrowCallback(Arc<Mutex<Inner>>);
 
 impl ThrowCallback {
-    pub(super) fn throw(&self, exc: &PyAny) {
+    pub(super) fn throw(&self, exc: Bound<'_, PyAny>) {
```

Comment:
Does it make sense to even just go for:

```suggestion
    pub(super) fn throw(&self, exc: Py<PyAny>) {
```

That'd avoid the need to `.unbind()` below, and maybe it also avoids the need to bind the object on its way in.